context_test.go

```diff
@@ -373,3 +347,49 @@ func TestCaseInsentitiveGOPATH(t *testing.T) {
 		t.Fatalf("expected %s, got %s", ip, pr)
 	}
 }
+
+func TestResolveProjectRoot(t *testing.T) {
+	tg := test.NewHelper(t)
+	defer tg.Cleanup()
+
+	tg.TempDir("go")
+	tg.TempDir("go/src")
+	tg.TempDir("sym")
+	tg.TempDir("go/src/real")
+	tg.TempDir("go/src/real/path")
+	tg.TempDir("go/src/sym")
+
+	tg.Setenv("GOPATH", tg.Path(filepath.Join(".", "go")))
+
+	ctx := &Ctx{GOPATH: tg.Path(filepath.Join(".", "go"))}
+
+	realPath := filepath.Join(ctx.GOPATH, "src/real/path")
+	symlinkedPath := filepath.Join(tg.Path("."), "sym", "symlink")
+	symlinkedInGoPath := filepath.Join(ctx.GOPATH, "src/sym/path")
+	os.Symlink(realPath, symlinkedPath)
+	os.Symlink(realPath, symlinkedInGoPath)
+
+	// Real path should be returned, no symlinks to deal with
+	p, err := ctx.resolveProjectRoot(realPath)
```

Comment:
nit: as a general naming pattern, we prefer "want" to "expect" - #152. That could be applied to both the var name here, and the wording of the error message below.